hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/ArrayListStreamSerializer.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.serialization.impl;
+
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.StreamSerializer;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+/**
+ * The {@link java.util.ArrayList} serializer
+ */
+public class ArrayListStreamSerializer
+        implements StreamSerializer<ArrayList> {
+
+    @Override
+    public void write(ObjectDataOutput out, ArrayList object)
+            throws IOException {
+        out.writeBoolean(object != null);
+        if (object != null) {
+            int size = object.size();
+            out.writeInt(size);
+            for (int i = 0; i < size; i++) {
+                out.writeObject(object.get(i));
+            }
+        }
+    }
+
+    @Override
+    public ArrayList read(ObjectDataInput in)
+            throws IOException {
+        if (in.readBoolean()) {
```

Comment:
It is a flavor thing but I would always load this in a variable 'notNull' or something.

boolean notNull = in.readBoolean();
if(notNull){...}

Now the code becomes self documenting.

Apart from that I would flip the flag. So write true in case of null, since afaik that is the most used approach in our code I believe.